### PR TITLE
Move SmartSim v0.X.Y test to `tests/_legacy` dir

### PR DESCRIPTION
**Pull request by MattToast**
Move the test suite used for SmartSim v0.X.Y to a `tests/_legacy` dir. This directory is explicitly skipped during test collection as these tests are not expected to pass as SmartSim transitions to a new user facing and core API.

### CODE REVIEW

File: doc/changelog.md [hunk 1]
```diff
@@ -16,10 +16,16 @@ To be released at some future point in time
 Description
 
 - Update packaging dependency
```
Comment (amandarichardsonn) [inline doc/changelog.md:18]: I would take out the changelog update, just bc we are not editing the changelog for PRs going into smartsim-refactor!

File: doc/api/smartsim_api.rst [hunk 2]
```diff
@@ -121,7 +120,6 @@ steps to a batch.
     SrunSettings.set_hostlist
     SrunSettings.set_excluded_hosts
     SrunSettings.set_cpus_per_task
-    SrunSettings.add_exe_args
```
Comment (amandarichardsonn) [inline doc/api/smartsim_api.rst:124]: nice catch!

**Review by amandarichardsonn — APPROVED** [review]
thank u for doing this!!

File: doc/changelog.md (hunk 1 above)
Comment (MattToast) [reply doc/changelog.md:18]: Good point! I'll just opt the the `smartsim-refactor` branch out of the CI check

File: doc/api/smartsim_api.rst (hunk 2 above)
Comment (MattToast) [reply doc/api/smartsim_api.rst:124]: Thank you for the the RTD CI action!

**Pull request merged**